### PR TITLE
B2sum: add -l/--length option

### PR DESCRIPTION
**Pull request by spineki**
# Summary
This pull request adds support for the -l/--length option to b2sum (hashsum).
B2sum supports variable output size if 
- length is a multiple of 8 bits
- length is less or equal than 512

## Related Issue
#4610 

## Remarks
This is my first attempt to contribute to this project, so I may have missed something. (same goes for naming conventions) !

I added a new function to allow creating a Blake2d instance with custom output size (`with_output_bytes`). I updated the reset function to use this new function instead of calling ::new().
Eventually, the Blake2d tuple has now a new parameter to keep track of the custom size. 

I had to create a new parsing step `uu_app_opt_length` that behaves more or less like `uu_app_opt_bits` but cannot share the same code due to different text / names.

## Tests
I added a simple test with the --length option enabled. I took example of the sha1 tests in the very same file.

related test:  `bash util/run-gnu-test.sh   tests/misc/b2sum-bsd.sh ` (-l flags)
Unfortunately, I only have Windows for now and I couldn't run the linux tests.

### CODE REVIEW

**Comment by sylvestre** [comment]
not your fault but sharing for other reviewers. the test doesn't pass because of:
```

2023-04-11T08:23:08.2374314Z error: unexpected argument '-c' found
2023-04-11T08:23:08.2374320Z 
2023-04-11T08:23:08.2374477Z   tip: to pass '-c' as a value, use '-- -c'
2023-04-11T08:23:08.2374482Z 
2023-04-11T08:23:08.2374575Z Usage: cksum [OPTIONS] [FILE]...
2023-04-11T08:23:08.2374580Z 
2023-04-11T08:23:08.2374724Z For more information, try '--help'.
2023-04-11T08:23:08.2374833Z FAIL tests/misc/b2sum.sh (exit status: 1)
```

File: src/uu/hashsum/src/hashsum.rs [hunk 1]
```diff
@@ -65,7 +65,32 @@ fn detect_algo(
...
+        "b2sum" => match matches.get_one::<usize>("length") {
+            Some(length_in_bits) => {
+                if *length_in_bits > 512 {
+                    crash!(1, "Invalid length (maximum digest length is 512 bits)")
+                }
+
+                // --length=0 falls back to default behavior
+                if *length_in_bits == 0 {
+                    return ("BLAKE2", Box::new(Blake2b::new()) as Box<dyn Digest>, 512);
+                }
```
Comment (cakebaker) [inline src/uu/hashsum/src/hashsum.rs:77]: I would move this functionality to its own arm `Some(0)`.

Comment (spineki) [reply src/uu/hashsum/src/hashsum.rs]: I combined it with `None` arm in commit fdbebfa

**Comment by cakebaker** [comment]
Thanks, great work :)

**Pull request merged**